src/dowel/csv_output.py

```diff
@@ -56,6 +65,32 @@ def record(self, data, prefix=''):
         else:
             raise ValueError('Unacceptable type.')
 
+    def dump(self, step=None):
+        super().dump()
+        if len(self._file_sections) > 1:
+            self._merge_file_sections()
+            self._file_sections = self._file_sections[:1]
+
+    def _merge_file_sections(self):
+        if len(self._file_sections) == 1:
+            return
+        os.rename(self._file_name, self._file_sections[0])
+        self._log_file = open(self._file_name, 'w')
+        self._init_writer()
+        self._writer.writeheader()
+        for sec_file_name in self._file_sections:
+            with open(sec_file_name)as f:
+                reader = csv.DictReader(f)
+                for r in reader:
+                    self._writer.writerow(r)
+            os.remove(sec_file_name)
+
+    def _init_writer(self):
+        self._writer = csv.DictWriter(
+            self._log_file,
+            fieldnames=sorted(list(self._fieldnames)),
+            extrasaction='ignore')
```

Comment:
Since we *should* be accounting for all fields, it's probably better to change extrasaction to `'raise'`.